response_operations_ui/views/messages.py

```diff
@@ -126,7 +127,64 @@ def view_messages():
         refined_messages = [_refine(msg) for msg in message_controllers.get_thread_list(params)]
         return render_template("messages.html", breadcrumbs=breadcrumbs, messages=refined_messages)
     except NoMessagesError:
-        return render_template("messages.html", breadcrumbs=breadcrumbs, response_error=True)
+        return render_template("messages.html",
+                               breadcrumbs=breadcrumbs,
+                               response_error=True)
+
+
+@messages_bp.route('/select-survey', methods=['GET', 'POST'])
+@login_required
+def view_select_survey():
+    breadcrumbs = [{"title": "Messages", "link": "/messages"},
+                   {"title": "Filter by survey"}]
+
+    if request.method == 'POST':
+        selected_survey = request.form.get('radio-answer', '')
+        return redirect(url_for("messages_bp.view_selected_survey",
+                                selected_survey=selected_survey))
+    else:
+        survey_list = [survey.value for survey in Surveys]
+        return render_template("message_select_survey.html",
+                               breadcrumbs=breadcrumbs,
+                               selected_survey=None,
+                               survey_list=survey_list)
+
+
+@messages_bp.route('/<selected_survey>', methods=['GET'])
+@login_required
+def view_selected_survey(selected_survey):
+    breadcrumbs = [{"title": selected_survey + " Messages"}]
+
+    params = {
+        'label': request.args.get('label'),
+        'survey': request.args.get('survey')
+    }
+
+    try:
+        refined_messages = [_refine(message) for message in message_controllers.get_thread_list(params)]
+        if selected_survey == 'FDI':
+            survey_id = _get_FDI_survey_id()
+        else:
+            survey_id = _get_survey_id(selected_survey)
+
+        filtered_messages = [messages for messages in refined_messages if messages['survey_id'] == survey_id]
```

Comment:
Is this right? It reads really weird to me.  Comprehensions are `[ expression for item in list if conditional ]`
but I don't get what the expression part of this one is doing.  Should the 2nd messages  actually be called 'message'?
The one above is clear, it's refining each message in the thread list.  It could also be me being dumb >_<